web/client/api/GeoStoreDAO.js

```diff
@@ -65,6 +66,28 @@ var Api = {
             return response.data;
         });
     },
+    login: function(username, password, options) {
+        let url = "session/login";
+        let authData;
+        return axios.post(url, null, this.addBaseUrl(_.merge({
+            auth: {
+                username: username,
+                password: password
+            }
+        }, options))).then((response) => {
+            authData = response.data;
+            return axios.get("users/user/details", this.addBaseUrl(_.merge({
+                headers: {
+                    'Authorization': 'Bearer ' + response.data.access_token
+                },
+                params: {
+                    includeattributes: true
+                }
+            }, options)));
+        }).then((response) => {
+            return { ...response.data, ...authData};
+        });
```

Comment:
Errors are catched in the action, see https://github.com/geosolutions-it/MapStore2/pull/1949/files#diff-cb95a3cd779691bf0b73f9f8eabed903L66